oak_crypto/src/noise_nk/rustcrypto.rs

```diff
@@ -0,0 +1,212 @@
+    }
+
+    pub fn compute_public_key(&self) -> [u8; P256_X962_LENGTH] {
+        p256::ProjectivePoint::mul_by_generator(&self.v)
+            .to_encoded_point(false)
+            .as_bytes()
+            .try_into()
+            .unwrap()
+    }
+
+    pub fn bytes(&self) -> [u8; P256_SCALAR_LENGTH] {
+        self.v.to_repr().as_slice().try_into().unwrap()
+    }
+}
+
+impl TryFrom<&[u8]> for P256Scalar {
+    type Error = ();
+
+    fn try_from(bytes: &[u8]) -> Result<Self, ()> {
+        let array: [u8; P256_SCALAR_LENGTH] = bytes.try_into().map_err(|_| ())?;
+        (&array).try_into()
+    }
+}
+
+impl TryFrom<&[u8; P256_SCALAR_LENGTH]> for P256Scalar {
+    type Error = ();
+
+    fn try_from(bytes: &[u8; P256_SCALAR_LENGTH]) -> Result<Self, ()> {
+        let scalar = p256::Scalar::from_repr((*bytes).into());
+        if !bool::from(scalar.is_some()) {
+            return Err(());
+        }
+        let scalar = scalar.unwrap();
+        if scalar.is_zero_vartime() {
+            return Err(());
+        }
+        Ok(P256Scalar { v: scalar })
+    }
+}
+
+pub fn p256_scalar_mult(
+    scalar: &P256Scalar,
+    point: &[u8; P256_X962_LENGTH],
+) -> Result<[u8; 32], ()> {
+    let point = p256::EncodedPoint::from_bytes(point).map_err(|_| ())?;
+    let affine_point = p256::AffinePoint::from_encoded_point(&point);
+    if !bool::from(affine_point.is_some()) {
+        // The peer's point is considered public input and so we don't need to work in
+        // constant time.
+        return Err(());
+    }
+    // unwrap: `is_some` checked just above.
+    let result = affine_point.unwrap().mul(scalar.v).to_encoded_point(false);
+    let x = result.x().ok_or(())?;
+    // unwrap: the length of a P256 field-element had better be 32 bytes.
+    Ok(x.as_slice().try_into().unwrap())
+}
+
+pub struct EcdsaKeyPair {
+    key_pair: p256::ecdsa::SigningKey,
+}
+
+impl EcdsaKeyPair {
+    pub fn from_pkcs8(pkcs8: &[u8]) -> Result<EcdsaKeyPair, ()> {
+        let key_pair: p256::ecdsa::SigningKey =
+            p256::ecdsa::SigningKey::from_pkcs8_der(pkcs8).map_err(|_| ())?;
+        Ok(EcdsaKeyPair { key_pair })
+    }
+
+    pub fn generate_pkcs8() -> Result<impl AsRef<[u8]>, ()> {
+        // WARNING: not actually a random scalar
+        let mut scalar = [0u8; P256_SCALAR_LENGTH];
+        scalar[0] = 42;
+        let non_zero_scalar = p256::NonZeroScalar::from_repr(scalar.into()).unwrap();
+        let key = p256::ecdsa::SigningKey::from(non_zero_scalar);
+        Ok(key.to_pkcs8_der().map_err(|_| ())?.to_bytes())
+    }
+
+    pub fn public_key(&self) -> impl AsRef<[u8]> + '_ {
+        p256::ecdsa::VerifyingKey::from(&self.key_pair).to_sec1_bytes()
+    }
+
+    pub fn sign(&self, signed_data: &[u8]) -> Result<impl AsRef<[u8]>, ()> {
+        let sig: ecdsa::Signature<p256::NistP256> = self.key_pair.sign(signed_data);
+        Ok(sig.to_der())
+    }
+}
+
+pub fn ecdsa_verify(pub_key: &[u8], signed_data: &[u8], signature: &[u8]) -> bool {
```

Comment:
This seems quite easy to misuse. How about returning `Result<(), ()>` (I know it looks weird), or at least annotate with [must_use](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-must_use-attribute)